app/forms/candidate_interface/science_gcse_grade_form.rb

```diff
@@ -203,7 +203,9 @@ def grade_hash(grade)
     end
 
     def sanitize(grade)
-      if DOUBLE_GCSE_GRADES.exclude?(grade)
+      if ALL_GCSE_GRADES.exclude?(grade) && grade_contains_number?(grade)
+        grade&.gsub(/[^%\d]/, '')&.insert(1, '-')
```

Comment:
Maybe something like this? Just so people don't have to figure out what's going on with the regex/insert 
```
add_dash_between_digits(grade)

def add_dash_between_digits(grade)
  grade&.gsub(/[^%\d]/, '')&.insert(1, '-')
end 

```